app/main.jsx

```diff
@@ -19,10 +19,16 @@ import AppContainer from './containers/AppContainer'
 import DreamsContainer from './containers/DreamsContainer'
 import AnalyticsContainer from './containers/AnalyticsContainer'
 
-function onSingleDreamEnter(nextRouterState){
+function onSingleDreamEnter(nextRouterState, replace, done){
   const dreamId = nextRouterState.params.id
-  const user=store.getState().auth
-  store.dispatch(fetchSingleDream(user.id,dreamId))
+  const user = store.getState().auth
+
+  //had to put axios call here so we can use the done function that comes with onEnter hooks. So the hook knows to not mount the component until this entire fn is done running.
+  axios.get(`/api/dreams/user/${user.id}/${dreamId}`)
+    .then(res => res.data)
+    .then(dream => store.dispatch(selectDream(dream)))
+    .then(() => done())
+    .catch(console.error)
 }
```

Comment:
omggg I didnt even know you could do this! solves so many async issues! brilliant.